dask/dataframe/groupby.py

```diff
@@ -411,6 +702,63 @@ def __getattr__(self, key):
         except KeyError as e:
             raise AttributeError(e)
 
+    @derived_from(pd.core.groupby.DataFrameGroupBy)
+    def aggregate(self, arg, split_every=None):
+        if arg == 'size':
+            return self.size()
+
+        if isinstance(self.index, tuple) or np.isscalar(self.index):
+            group_columns = {self.index}
+
+        elif isinstance(self.index, list):
+            group_columns = {i for i in self.index
+                             if isinstance(i, tuple) or np.isscalar(i)}
+
+        else:
+            group_columns = set()
+
+        # NOTE: this step relies on the index normalization to replace series
+        #       with their name in an index.
+        non_group_columns = [col for col in self.obj.columns
+                             if col not in group_columns]
+        spec = _normalize_spec(arg, non_group_columns)
+        chunk_funcs, aggregate_funcs, finalizers = _build_agg_args(spec)
+
+        if isinstance(self.index, (tuple, list)) and len(self.index) > 1:
+            levels = list(range(len(self.index)))
+        else:
+            levels = 0
+
+        # TODO: add normed spec as an additional part to the token?
+        token = 'aggregate'
+        finalize_token = '{}-finalize'.format(token)
```

Comment:
I assumed additional logic was required. Since this is not the case, I removed both tokens and inlined them directly into the `aca` and `map_partitions` calls.